pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/BaseClusterIntegrationTest.java

```diff
@@ -78,303 +99,478 @@
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
-import com.google.common.util.concurrent.AsyncFunction;
-import com.google.common.util.concurrent.Futures;
-import com.google.common.util.concurrent.ListenableFuture;
-import com.google.common.util.concurrent.ListenableFutureTask;
-import com.google.common.util.concurrent.Uninterruptibles;
-import com.linkedin.pinot.client.ConnectionFactory;
-import com.linkedin.pinot.common.utils.EqualityUtils;
-import com.linkedin.pinot.common.utils.KafkaStarterUtils;
-import com.linkedin.pinot.common.utils.StringUtil;
-import com.linkedin.pinot.common.utils.TarGzCompressionUtils;
-import com.linkedin.pinot.common.utils.ZkStarter;
-import com.linkedin.pinot.core.indexsegment.generator.SegmentGeneratorConfig;
-import com.linkedin.pinot.core.indexsegment.utils.AvroUtils;
-import com.linkedin.pinot.core.segment.creator.SegmentIndexCreationDriver;
-import com.linkedin.pinot.core.segment.creator.impl.SegmentCreationDriverFactory;
-import com.linkedin.pinot.server.util.SegmentTestUtils;
-import com.linkedin.pinot.util.TestUtils;
-import kafka.javaapi.producer.Producer;
-import kafka.producer.KeyedMessage;
-import kafka.producer.ProducerConfig;
 
 
 /**
  * Shared implementation details of the cluster integration tests.
  */
 public abstract class BaseClusterIntegrationTest extends ClusterTest {
   private static final Logger LOGGER = LoggerFactory.getLogger(BaseClusterIntegrationTest.class);
+
   private static final AtomicInteger totalAvroRecordWrittenCount = new AtomicInteger(0);
   private static final boolean BATCH_KAFKA_MESSAGES = true;
   private static final int MAX_MESSAGES_PER_BATCH = 10000;
-  private static final int MAX_MULTIVALUE_CARDINALITY = 5;
-  protected static final boolean GATHER_FAILED_QUERIES = false;
 
-  private int failedQueryCount = 0;
-  private int queryCount = 0;
+  // Maximum number of queries skipped when random select queries from hard coded query set.
+  private static final int MAX_NUM_QUERIES_SKIPPED = 200;
```

Comment:
Currently the logic is not hard (see method testHardcodedQuerySet()), the benefits of this logic is it can save memory and prevent selecting the same query multiple times.